### ISSUE TITLE
Concurrency issue: The quota has been exceeded.

### ISSUE THREAD

**Issue by droganov** [nanostores/persistent#7]
When I have only 1 window the store works and I can see LS is updated:

```js
{"skipRules":false,"lastWizardSection":"title"} //1
{"skipRules":false,"lastWizardSection":"price"} //2
```

But when I open same page in another window of the same browser and try to change store I get:

- FF `QuotaExceededError: The quota has been exceeded.`
- Opera: `QuotaExceededError: Failed to set a named property on 'Storage': Setting the value of 'settings' exceeded the quota.`

And in the LS I can see my JSON became this:

FF:
"\"\\\"\\\\\\\"\\\\\\\\\\\\\\\"\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\"\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\"\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\"\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\"\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\

In opera there is a similar picture, but the string is much longer

The store source is:
```ts
import { createPersistentStore } from '@nanostores/persistent'
import { update } from 'nanostores'

import { useClientStore } from '../hooks/useClientStore'

type Settings = {
  lastWizardSection?: string
  skipRules?: boolean
}

const defaultSettings: Settings = { skipRules: false }

const settingsStore = createPersistentStore<Settings>(
  'settings',
  defaultSettings,
  {
    encode: JSON.stringify,
    decode(value) {
      try {
        return JSON.parse(value) as Settings
      } catch (error) {
        // todo: log error
        return defaultSettings
      }
    },
  }
)

export const patchSettings = (patch: Partial<Settings>): void => {
  update(settingsStore, (last) => ({ ...last, ...patch }))
}

export const useSettings = (): Settings =>
  useClientStore(settingsStore, defaultSettings)

```

```ts
import { ReadableStore } from 'nanostores'
import { useStore } from 'nanostores/react'
import { useIsMounted } from './useIsMounted'

export const useClientStore = <S>(
  store: ReadableStore<S>,
  defaultState: S
): S => {
  const isMounted = useIsMounted()
  const state = useStore(store)
  return isMounted ? state : defaultState
}

```

**Comment by ai** [comment]
Fixed 10c7b06

**Comment by ai** [comment]
Released in 0.3.2.
